### DISCUSSION TITLE
wire:model not working with alpine x-model

### DISCUSSION THREAD

**Discussion by jochemgruter** [livewire/livewire#2196] [1. Help] [answered]
Hi Guys,

I have the following line of code in my livewire component view:

`<input type="text" x-model="name" wire:model="name">`

When the component is rendered, the value from the laravel component is set correctly to the input. But when the value is changed from the alpine model, then livewire is remembering its old value. Only if I manually change the value, it's pushed to livewire.

How can I make it work that the value from the alpine model will be pushed to livewire?

**Comment by jskorlol** [comment, marked as answer]
https://laravel-livewire.com/docs/2.x/alpine-js#extracting-blade-components

Sharing State Between Livewire And Alpine: @entangle

**Reply by jochemgruter** [reply]
Thanks, this is working with the @engtable. The only thing is that when the value has changed, a server request is made, but this is not necessary. 

On my input I have used `wire:model.defer`, but it's still making a request on every value change.

**Reply by jochemgruter** [reply]
Ah, just use `@engtable('').defer` I read now.

**Comment by sentiasa** [comment]
It was because `x-model` was not triggering Change Event.

I got it working by this way:
```
<div x-data="{ checked: false }"
     x-init="
           $watch('checked', (val) => { 
               checked = val; 
               $refs.checkbox.checked = val; 
               $refs.checkbox.dispatchEvent(new Event('change'));
          })
     "
>
   <span @click="checked = !checked">Change checked</span>
   <input wire:model.defer="test" x-ref="checkbox" type="checkbox">
</div>
```
